### PR TITLE
Add support for 3.9, and fix build and add support for 3.10.

### LINKED ISSUE #58: Fails to build on Python 3.10.0a2
Trying to build zodbpickle on Python 3.10.0a2 fails with

```
  gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/home/mg/src/zodbbrowser/.tox/py310/include -I/home/mg/opt/python310/include/python3.10 -c src/zodbpickle/_pickle_33.c -o build/temp.linux-x86_64-3.10/src/zodbpickle/_pickle_33.o
  src/zodbpickle/_pickle_33.c: In function ‘Pdata_New’:
  src/zodbpickle/_pickle_33.c:179:19: error: lvalue required as left operand of assignment
    179 |     Py_SIZE(self) = 0;
        |                   ^
  src/zodbpickle/_pickle_33.c: In function ‘Pdata_clear’:
  src/zodbpickle/_pickle_33.c:205:19: error: lvalue required as left operand of assignment
    205 |     Py_SIZE(self) = clearto;
        |                   ^
  src/zodbpickle/_pickle_33.c: In function ‘Pdata_pop’:
  src/zodbpickle/_pickle_33.c:247:23: error: lvalue required as decrement operand
    247 |     return self->data[--Py_SIZE(self)];
        |                       ^~
  src/zodbpickle/_pickle_33.c: In function ‘Pdata_push’:
  src/zodbpickle/_pickle_33.c:257:29: error: lvalue required as increment operand
    257 |     self->data[Py_SIZE(self)++] = obj;
        |                             ^~
  src/zodbpickle/_pickle_33.c: In function ‘Pdata_poptuple’:
  src/zodbpickle/_pickle_33.c:283:19: error: lvalue required as left operand of assignment
    283 |     Py_SIZE(self) = start;
        |                   ^
  src/zodbpickle/_pickle_33.c: In function ‘Pdata_poplist’:
  src/zodbpickle/_pickle_33.c:300:19: error: lvalue required as left operand of assignment
    300 |     Py_SIZE(self) = start;
        |                   ^
  src/zodbpickle/_pickle_33.c: In function ‘save_long’:
  src/zodbpickle/_pickle_33.c:1646:18: warning: implicit declaration of function ‘_PyUnicode_AsStringAndSize’; did you mean ‘PyUnicode_FromStringAndSize’? [-Wimplicit-function-declaration]
   1646 |         string = _PyUnicode_AsStringAndSize(repr, &size);
        |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~
        |                  PyUnicode_FromStringAndSize
  src/zodbpickle/_pickle_33.c:1646:16: warning: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
   1646 |         string = _PyUnicode_AsStringAndSize(repr, &size);
        |                ^
  src/zodbpickle/_pickle_33.c: In function ‘save_pers’:
  src/zodbpickle/_pickle_33.c:2942:29: warning: assignment to ‘char *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
   2942 |             pid_ascii_bytes = _PyUnicode_AsStringAndSize(pid_str, &size);
        |                             ^
  src/zodbpickle/_pickle_33.c: In function ‘Pickler_init’:
  src/zodbpickle/_pickle_33.c:3582:9: warning: implicit declaration of function ‘_PyObject_HasAttrId’; did you mean ‘_PyObject_SetAttrId’? [-Wimplicit-function-declaration]
   3582 |     if (_PyObject_HasAttrId((PyObject *)self, &PyId_persistent_id)) {
        |         ^~~~~~~~~~~~~~~~~~~
        |         _PyObject_SetAttrId
  src/zodbpickle/_pickle_33.c: In function ‘load_pop’:
  src/zodbpickle/_pickle_33.c:4840:30: error: lvalue required as left operand of assignment
   4840 |         Py_SIZE(self->stack) = len;
        |                              ^
  src/zodbpickle/_pickle_33.c: In function ‘do_append’:
  src/zodbpickle/_pickle_33.c:5145:38: error: lvalue required as left operand of assignment
   5145 |                 Py_SIZE(self->stack) = x;
        |                                      ^
  src/zodbpickle/_pickle_33.c:5151:30: error: lvalue required as left operand of assignment
   5151 |         Py_SIZE(self->stack) = x;
        |                              ^
  src/zodbpickle/_pickle_33.c: In function ‘Pdata_pop’:
  src/zodbpickle/_pickle_33.c:248:1: warning: control reaches end of non-void function [-Wreturn-type]
    248 | }
        | ^
  error: command '/usr/lib/ccache/gcc' failed with exit code 1
```

### PR DESCRIPTION
**Pull request by jamadden**
Fixes #60 and fixes #58.

### CODE REVIEW

**Comment by jamadden** [comment]
This repo hasn't been updated to GitHub Actions yet, so we don't actually have a test environment for 3.10, but it does work locally for me.

```console
$ at 05:53:22 -> tox -re py310
py310 recreate: //zodbpickle/.tox/py310
py310 installdeps: coverage
py310 develop-inst: //zodbpickle
py310 installed: //.tox/py310/lib/python3.10/site-packages/pip/_vendor/packaging/tags.py:7: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives,  import distutils.util,/.tox/py310/lib/python3.10/site-packages/pip/_vendor/packaging/version.py:127: DeprecationWarning: Creating a LegacyVersion has been deprecated and will be removed in the next major release,  warnings.warn(,//3.10/lib/python3.10/distutils/command/install.py:13: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead,  from distutils.sysconfig import get_config_vars,coverage==5.6b1,-e git+https://github.com/zopefoundation/zodbpickle.git@20fb59630db28c8dbb06002e3a71fe77e08942e6#egg=zodbpickle
py310 run-test-pre: PYTHONHASHSEED='2646488306'
py310 run-test: commands[0] | coverage run setup.py -q test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
.......................................sss...............................................................................sss.........................................................................sss...............................................................sss.......................................................................sssss................sss.................................................................
----------------------------------------------------------------------
Ran 426 tests in 5.370s

OK (skipped=20)
_______________________________________ summary ________________________________________
  py310: commands succeeded
  congratulations :)

$ ./.tox/py310/bin/python --version
Python 3.10.0b1
```

**Review by icemac — COMMENTED** [review]
Tests run locally, but i cannot say anything about the code changes.

**Comment by icemac** [comment]
Rebased PR to `master` and updated to current `meta/config`.

**Comment by jamadden** [comment]
Neat 👍

**Comment by icemac** [comment]
Is there someone who could review this PR or is it good enough to be merged as it is?

**Comment by jamadden** [comment]
Neither the build nor test phases emit any warnings (well, from code we control anyway), and I have some projects that would like to support 3.10 but are waiting on a zodbpickle release, so this (obviously 😄) LGTM. But it's my PR, so I don't know that I get a vote.

File: src/zodbpickle/_pickle_33.c [hunk 1]
```diff
@@ -4,6 +4,28 @@
...
+/**
+ * The function ``_PyObject_LookupAttrId`` function replaces the combo of
+ * ``_PyObject_HasAttrId`` followed by ``_PyObject_GetAttrId``; our code isn't
+ * structured to take advantage of that, so for now we throw away the
+ * resulting attribute value and continue to make the extra ``Get`` call.
+ */
+static int _PyObject_HasAttrId(PyObject* obj, void* id)
+{
+    int result;
+    PyObject* attr_val;
```
Comment (jamadden) [inline src/zodbpickle/_pickle_33.c:22]: We're not initializing this to `NULL`, but that's not essential. All the error paths through `_PyObject_LookupAttrId` set it to `NULL` on failure, one way or the other, so it's always safe to call `Py_XDECREF`: `attr_val` will never hold garbage at that point. A fair amount of code in CPython itself depends on this behaviour so it's unlikely to change.

**Comment by icemac** [comment]
I just released https://pypi.org/project/zodbpickle/2.2.0/#files. If it breaks something, we can fix it later.

**Pull request merged**